src/rust/src/arrow_interop/mod.rs

```diff
@@ -61,3 +63,16 @@ impl RPackage for NanoArrowRPackage {
         "#)
     }
 }
+
+#[extendr]
+pub fn polars_allocate_array_stream() -> Robj {
+    let aas = Box::new(arrow::ffi::ArrowArrayStream::empty());
+    let x = Box::leak(aas); // leak box to make lifetime static
+    let x = x as *mut arrow::ffi::ArrowArrayStream;
```

Comment:
I'm curious, why not simply `let x = Box::into_raw(aas)`?

According to [this answer](https://stackoverflow.com/a/73330433/10194), it would be the prefferred idiom for FFI usecases such as this (in addition to being slightly shorter :-)).

Reply:
Thanks for your review!
I just moved and renamed this function from elsewhere in the codebase here and did not inspect the contents.
The change seems to be fine.